### ISSUE TITLE
Channel renaming

### ISSUE THREAD

**Issue by gabm** [mamba-org/quetz#536]
Currently the channel names are fixed as they are used as primary keys/ids for everything else in the DB. Would it be possible to use the current channel_name as "channel_id" and fill it with a fixed UUID for internal cross-references and add a new "channel_name" column for the public interface?

This way we could safely provide an API to rename channels...

**Comment by wolfv** [comment]
Hi @gabm, thanks for the suggestion. 
We actually do have the same problem with the package store (where we also use the channel name as "folder" to store the files in).
So it would break a couple of assumptions that are in the code base currently.

I am also wondering how often (in practice) channels would be renamed... it _should_ ideally happen fairly seldomly :)

**Comment by gabm** [comment]
> I am also wondering how often (in practice) channels would be renamed... it should ideally happen fairly seldomly :)

yes.. but making it impossible in the first place seems overly restrictive :)

So I understand, at least for the time being, there is no interest in such a patch?

**Comment by wolfv** [comment]
Certainly there is an interest in such a patch but it will be tricky.

– existing instances need to migrate their data store
– we need to make sure that performance is similar
– the database will have to be migrated

I just think that we have other subjects in the quetz code base that need fixing more urgently :) But if you want to tackle this, feel free!
